test/Common/DurableHttpTests.cs

```diff
@@ -120,6 +121,106 @@ public void DeserializeCallActivity()
             Assert.Empty(customHeaderValues);
         }
 
+        [Fact]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        public void SerializeManagedIdentityOptions()
+        {
+            // Part 1: Check if ManagedIdentityOptions is correctly serialized with TestDurableHttRequest
+            Dictionary<string, string> headers = new Dictionary<string, string>();
+            headers.Add("Accept", "application/json");
+
+            ManagedIdentityOptions options = new ManagedIdentityOptions();
+            options.AuthorityHost = new Uri("https://dummy.login.microsoftonline.com/");
+            options.TenantId = "tenant_id";
+
+            MockTokenSource mockTokenSource = new MockTokenSource("dummy token", options);
+
+            TestDurableHttpRequest request = new TestDurableHttpRequest(
+                httpMethod: HttpMethod.Get,
+                headers: headers,
+                tokenSource: mockTokenSource);
+
+            string serializedTestDurableHttpRequest = JsonConvert.SerializeObject(request);
+
+            Assert.Contains("\"authorityhost\":\"https://dummy.login.microsoftonline.com/", serializedTestDurableHttpRequest);
```

Comment:
For slightly more rigorous testing, we could build up the whole expected JSON object as a string, and do something like `Assert.True(JToken.DeepEquals(JObject.Parse(expectedJson), JObject.Parse(serializedTestDurableHttpRequest))`